core/arch/arm/plat-sam/drivers/clk/at91_pll.c

```diff
@@ -0,0 +1,303 @@
+// SPDX-License-Identifier: GPL-2.0-or-later or BSD-3-Clause
+/*
+ *  Copyright (C) 2013 Boris BREZILLON <b.brezillon@overkiz.com>
+ *  Copyright (C) 2021 Microchip
+ */
+
+#include <io.h>
+#include <kernel/delay.h>
+#include <kernel/panic.h>
+#include <util.h>
+#include <mm/core_memprot.h>
+#include <types_ext.h>
+
+#include "at91_clk.h"
+
+#define PLL_STATUS_MASK(id)	(1 << (1 + (id)))
+#define PLL_REG(id)		(AT91_CKGR_PLLAR + ((id) * 4))
+#define PLL_DIV_MASK		0xff
+#define PLL_DIV_MAX		PLL_DIV_MASK
+#define PLL_DIV(reg)		((reg) & PLL_DIV_MASK)
+#define PLL_MUL(reg, layout)	(((reg) >> (layout)->mul_shift) & \
+				 (layout)->mul_mask)
+#define PLL_MUL_MIN		2
+#define PLL_MUL_MASK(layout)	((layout)->mul_mask)
+#define PLL_MUL_MAX(layout)	(PLL_MUL_MASK(layout) + 1)
+#define PLL_ICPR_SHIFT(id)	((id) * 16)
+#define PLL_ICPR_MASK(id)	(0xffff << PLL_ICPR_SHIFT(id))
+#define PLL_MAX_COUNT		0x3f
+#define PLL_COUNT_SHIFT		8
+#define PLL_OUT_SHIFT		14
+#define PLL_MAX_ID		1
+
+struct clk_pll {
+	vaddr_t base;
+	uint8_t id;
+	uint8_t div;
+	uint8_t range;
+	uint16_t mul;
+	const struct clk_pll_layout *layout;
+	const struct clk_pll_characteristics *characteristics;
+};
+
+static inline bool clk_pll_ready(vaddr_t base, int id)
+{
+	unsigned int status = io_read32(base + AT91_PMC_SR);
+
+	return status & PLL_STATUS_MASK(id) ? 1 : 0;
+}
+
+static TEE_Result clk_pll_enable(struct clk *clk)
+{
+	struct clk_pll *pll = clk->priv;
+	const struct clk_pll_layout *layout = pll->layout;
+	const struct clk_pll_characteristics *characteristics =
+							pll->characteristics;
+	uint8_t id = pll->id;
+	uint32_t mask = PLL_STATUS_MASK(id);
+	int offset = PLL_REG(id);
+	uint8_t out = 0;
+	unsigned int pllr = 0;
+	unsigned int status = 0;
+	uint8_t div = 0;
+	uint16_t mul = 0;
+
+	pllr = io_read32(pll->base + offset);
+	div = PLL_DIV(pllr);
+	mul = PLL_MUL(pllr, layout);
+
+	status = io_read32(pll->base + AT91_PMC_SR);
+	if ((status & mask) &&
+	    (div == pll->div && mul == pll->mul))
```

Comment:
fits in a single line, maybe
```c
	if (status & mask && div == pll->div && mul == pll->mul)
		return TEE_SUCCESS;
```

Reply:
Honestly, I'm not sure this is more readable.